contracts/ethereum/src/core/node-codec/NodeBuilder.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.17;
+
+import "../../interfaces/ICodec.sol";
+import "../../interfaces/ITrie.sol";
+
+contract NodeBuilder is ICodec, ITrie {
+    NodeHandle[] emptyChildren;
+    NodeHandle emptyChild;
+    Slice emptySlice;
+    Value emptyValue;
+
+    uint8 constant NIBBLE_LENGTH = 16;
+
+    function buildNode(NodePlanStruct memory nodePlan, uint8[] calldata data)
```

Comment:
can you please add more docs on the functions?

Reply:
done